include/stdexec/execution.hpp

```diff
@@ -5485,6 +5485,203 @@ namespace stdexec {
   using __when_all::transfer_when_all_with_variant_t;
   inline constexpr transfer_when_all_with_variant_t transfer_when_all_with_variant{};
 
+  namespace __when_any_ {
+  template <class _BaseEnv>
+  using __env_t =
+      __make_env_t<_BaseEnv, __with<get_stop_token_t, in_place_stop_token>>;
+
+  template <class _Receiver, class _Op> struct __receiver_id {
```

Comment:
Template heads on a separate line, pls. Here and elsewhere.

```suggestion
  template <class _Receiver, class _Op>
    struct __receiver_id {
```

And although I like the naming scheme of the `_id` suffix for class templates that are ids, the point of the feature is to make the type names read well. A type made up of things like `__foo::__sender` and `__foo::__receiver` will read better than if they were `__foo::__sender_id` and `__foo::__receiver_id`.

Finally, I have all sorts of trouble when trying to parameterize the receiver on the type of the operation state. It's too easy to get into a situation where contraints on the receiver's methods require the type of the operation state type to be complete, or ending up with unfathomable recursion in constraints.

What I generally do instead is to factor out into a base class the part of the op state that the receiver needs to know about. This is usually everything except the inner operation state(s). Then the receiver stores a pointer to the base, keeping cycles out of the type system.